packages/system-access/src/base/credentials.ts

```diff
@@ -0,0 +1,125 @@
+}
+
+/**
+ * Check the secure storage if it has credentials for the given target.
+ *
+ * @param target ABAP target
+ * @param logger - reference to the logger instance
+ * @returns credentials from the store or undefined.
+ */
+export async function getCredentialsFromStore(
+    target: UrlAbapTarget,
+    logger: Logger
+): Promise<BackendSystem | undefined> {
+    try {
+        if (!isAppStudio()) {
+            const systemService = await getService<BackendSystem, BackendSystemKey>({ entityName: 'system' });
+            let system = await systemService.read(new BackendSystemKey({ url: target.url, client: target.client }));
+            // check if there are credentials for the default client
+            if (!system && target.client) {
+                system = await systemService.read(new BackendSystemKey({ url: target.url }));
+            }
+            return system;
+        }
+    } catch (error) {
+        logger.warn('Reading credentials from store failed');
+        logger.debug(error.message);
+    }
+    return undefined;
+}
+
+/**
+ * Store the credentials in the secure storage.
+ *
+ * @param name system name
+ * @param target target
+ * @param target.url system url
+ * @param target.client optional system client
+ * @param credentials basic auth credentials
+ * @param credentials.username username
+ * @param credentials.password password
+ * @param logger reference to the logger instance
+ * @returns true if the credentials are successfully stored
+ */
+export async function storeCredentials(
+    name: string,
+    target: { url: string; client?: string },
+    credentials: { username: string; password: string },
+    logger: Logger
+): Promise<boolean> {
+    try {
+        const systemService = await getService<BackendSystem, BackendSystemKey>({ entityName: 'system' });
+        const system = new BackendSystem({
+            name,
+            ...target,
+            ...credentials
+        });
+        await systemService.write(system);
+        return true;
+    } catch (error) {
+        logger.error('Could not store credentials.');
+        logger.debug(error);
+        return false;
+    }
+}
+
+/**
+ * Checks the environment variables for Fiori tools settings.
+ *
+ * @returns basic auth credentials from the environment or undefined.
+ */
+export function getCredentialsFromEnvVariables(): BasicAuth | undefined {
+    if (process.env.FIORI_TOOLS_USER && process.env.FIORI_TOOLS_PASSWORD) {
+        return {
+            username: process.env.FIORI_TOOLS_USER,
+            password: process.env.FIORI_TOOLS_PASSWORD
+        };
+    } else {
+        return undefined;
+    }
+}
+
+/**
+ * Prompt for username and password.
+ *
+ * @param username - optional username that is to be offered as default
+ * @returns credentials object with username/password
+ */
+export async function getCredentialsWithPrompts(username?: string): Promise<BasicAuth> {
```

Comment:
@cianmSAP this is something we can consume back into ui5-tooling to handle the credentials prompt?

Reply:
Yes definitely @longieirl